### DISCUSSION TITLE
solution for Cannot find module 'nx/src/plugins/js/lock-file/lock-file

### DISCUSSION THREAD

**Discussion by jmls** [nrwl/nx#16345] [General]
a FYI : I was getting this error when trying to generate a plugin using this command line

` $ nx g @nrwl/nx-plugin:plugin`

![image](https://user-images.githubusercontent.com/929523/232527944-a48cd8c9-a936-4bf2-ab40-5f3f12550b3b.png)

and it obviously fails

I  tried removing node_modules 

I saw that the global and local versions of nx (nx --version) were different , and migrated to the latest version (15.9.2) 

This seems to have fixed the issue. At least it runs now ;)

**Comment by marko-hologram** [comment]
NX always suggests running the same version of all their tools because that guarantees that these tools were tested together and should work. If you have a version mismatch (e.g. nx on 15.1.0, but @nrwl/next on 15.4.0), that can cause issues.

So in your project, you should ideally have all nx/@nrwl dependencies on the same version and then to use local (in project) version of NX, run commands using `npx nx ...` or `yarn nx ...` or `pnpm nx ...` to ensure that NX installed in that project is used for any NX command you run.

**Comment by jmls** [comment]
I agree - suppose I was surprised that nx didn't highlight this instead of throwing some odd error ;)
